ibm/service/power/data_source_ibm_pi_cloud_connection.go

```diff
@@ -7,129 +7,127 @@ import (
+				Type:        schema.TypeString,
 			},
-			PICloudConnectionIBMIPAddress: {
-				Type:     schema.TypeString,
-				Computed: true,
+			Attr_GreSourceAddress: {
+				Computed:    true,
+				Description: "GRE auto-assigned source IP address.",
+				Type:        schema.TypeString,
 			},
-			PICloudConnectionUserIPAddress: {
-				Type:     schema.TypeString,
-				Computed: true,
+			Attr_IBMIPAddress: {
+				Computed:    true,
+				Description: "The IBM IP address.",
+				Type:        schema.TypeString,
 			},
-			PICloudConnectionPort: {
-				Type:     schema.TypeString,
-				Computed: true,
+			Attr_Metered: {
+				Computed:    true,
+				Description: "Enable metering for this cloud connection.",
+				Type:        schema.TypeBool,
 			},
-			PICloudConnectionNetworks: {
-				Type:        schema.TypeSet,
+			Attr_Networks: {
 				Computed:    true,
-				Elem:        &schema.Schema{Type: schema.TypeString},
 				Description: "Set of Networks attached to this cloud connection",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Type:        schema.TypeSet,
 			},
-			PICloudConnectionClassicEnabled: {
-				Type:        schema.TypeBool,
+			Attr_Port: {
 				Computed:    true,
-				Description: "Enable classic endpoint destination",
-			},
-			PICloudConnectionClassicGreDest: {
+				Description: "Port.",
 				Type:        schema.TypeString,
+			},
+			Attr_Speed: {
 				Computed:    true,
-				Description: "GRE destination IP address",
+				Description: "Speed of the cloud connection (speed in megabits per second)",
+				Type:        schema.TypeInt,
 			},
-			PICloudConnectionClassicGreSource: {
-				Type:        schema.TypeString,
+			Attr_Status: {
 				Computed:    true,
-				Description: "GRE auto-assigned source IP address",
+				Description: "Link status.",
+				Type:        schema.TypeString,
 			},
-			PICloudConnectionVPCEnabled: {
-				Type:        schema.TypeBool,
+			Attr_UserIPAddress: {
 				Computed:    true,
-				Description: "Enable VPC for this cloud connection",
+				Description: "User IP address.",
+				Type:        schema.TypeString,
 			},
-			PICloudConnectionVPCCRNs: {
-				Type:        schema.TypeSet,
+			Attr_VPCCRNs: {
 				Computed:    true,
+				Description: "Set of VPCs attached to this cloud connection.",
 				Elem:        &schema.Schema{Type: schema.TypeString},
-				Description: "Set of VPCs attached to this cloud connection",
+				Type:        schema.TypeSet,
 			},
-			PICloudConnectionConnectionMode: {
-				Type:        schema.TypeString,
+			Attr_VPCEnabled: {
 				Computed:    true,
-				Description: "Type of service the gateway is attached to",
+				Description: "Enable VPC for this cloud connection.",
+				Type:        schema.TypeBool,
 			},
 		},
 	}
 }
 
-func dataSourceIBMPICloudConnectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func dataSourceIBMAttr_Read(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
I think I find and replace error on my part accidentally renamed the name of this function.

I should be dataSourceIBMPICloudConnectionRead.

Reply:
Changes made